### DISCUSSION TITLE
Location Fuzzing Vulnerability

### DISCUSSION THREAD

**Discussion by snake0073** [meshtastic/Discussions#110] [General]
### Summary
While utilizing imprecise location _with_ an enabled GPS providing location data, the actual location of the node can be determined by using the Node Map feature in the client app.

### Details
T-beam firmware 2.5.14.f2ee0df
App version  2.5.12 Android

### PoC
Setup a node with an enabled GPS, apply an imprecise location amount in the channel settings, and leave it static for a period of time. Eventually, the historical location points on the node map will intersect enough to determine the actual location with reasonable certainty. 

### Impact
Location fuzzing doesn't really hide you with an enabled GPS. 

See attached screenshots: 
![Screenshot_20241204-154031](https://github.com/user-attachments/assets/d7458861-9aa0-47d4-9e6f-b4def0468f76)
![Screenshot_20241204-154430~2](https://github.com/user-attachments/assets/a538010a-d2ce-4126-aa25-9158a21689ff)

**Comment by garthvh** [comment]
There should really not be history displayed or logged for imprecise positions, which is not available on iOS.  What numeric value is being used here?

**Reply by snake0073** [reply]
The first screenshot is one of my nodes, using the 299ft value. I can absolutely confirm I was at the center of the intersected lines. 

The second screenshot is someone else, not sure what value they had set.

**Comment by Soupbones** [comment]
I put in a fake fixed location then set the precision from that location. That basically makes it even less likely to be found. Only some decent radio direction finding equipment like a Kraken SDR will lt you find it.

**Comment by easytarget** [comment]
Position fuzzing does not work by picking some random 'other position nearby', my understanding was that it works by the meshtastic node dropping a (user selected) number bits from the end of the lat/long co-ordinate data that it transmits.
- This means that the reported positions will snap to a 'grid' with a size being dictated by how many bits are being removed.
- You can see this in the first screenshot. If the OP was in the exact center of the grid that is just a co-incidence, you could get the same result even if quite off-center.
- I'd ignore the second trace, that just appears to be normal, untrimmed, gps data. If it was properly fuzzed the points would be falling in a grid pattern. This is manifestly not the case.

However; the question of whether the app, on any platform, should be recording a history is not really relevant here.. If a bad actor wants to record reported GPS positions over time they will not be stopped just because the app does not do it.

**Reply by garthvh** [reply]
300 feet is the distance used in the image in question, which is not enough to expect the location to be obscured.
